packages/integrations/markdoc/README.md

```diff
@@ -0,0 +1,342 @@
+
+<Aside client:load />
+```
+
+This component [can be applied via the `components` prop](#render-html-elements-as-astro-components):
+
+```astro
+---
+// src/pages/why-markdoc.astro
+import { getEntryBySlug } from 'astro:content';
+import ClientAside from '../components/ClientAside.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    aside: ClientAside,
+  }}
+/>
+```
+
+### Markdoc config
+
+The Markdoc integration accepts [all Markdoc configuration options](https://markdoc.dev/docs/config), including [tags](https://markdoc.dev/docs/tags) and [functions](https://markdoc.dev/docs/functions).
+
+You can pass these options from the `markdoc()` integration in your `astro.config`. This example adds a global `getCountryEmoji` function:
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import markdoc from '@astrojs/markdoc';
+
+// https://astro.build/config
+export default defineConfig({
+  integrations: [
+    markdoc({
+      functions: {
+        getCountryEmoji: {
+          transform(parameters) {
+            const [country] = Object.values(parameters);
+            const countryToEmojiMap = {
+              japan: '🇯🇵',
+              spain: '🇪🇸',
+              france: '🇫🇷',
+            }
+            return countryToEmojiMap[country] ?? '🏳'
+          },
+        },
+      },
+    }),
+  ],
+});
+```
+
+Now, you can call this function from any Markdoc content entry:
+
+```md
+¡Hola {% getCountryEmoji("spain") %}!
+```
+
+:::note
+These options will be applied during [the Markdoc "transform" phase](https://markdoc.dev/docs/render#transform). This is run **at build time** (rather than server request time) both for static and SSR Astro projects. If you need to define configuration at runtime (ex. SSR variables), [see the next section](#Define-markdoc-configuration-at-runtime).
+:::
+
+📚 [See the Markdoc documentation](https://markdoc.dev/docs/functions#creating-a-custom-function) for more on using variables or functions in your content.
+
+### Define Markdoc configuration at runtime
+
+You may need to define Markdoc configuration at the component level, rather than the `astro.config.mjs` level. This is useful when mapping props and SSR parameters to [Markdoc variables](https://markdoc.dev/docs/variables).
+
+Astro recommends running the Markdoc transform step manually for this. First, install the `@markdoc/markdoc` package into your project:
+
+```sh
+# Using NPM
+npx astro add @markdoc/markdoc
+# Using Yarn
+yarn astro add @markdoc/markdoc
+# Using PNPM
+pnpm astro add @markdoc/markdoc
+```
+
+Then, call Markdoc's rendering functions where you plan to define your configuration. This example defines an `abTestGroup` Markdoc variable based on an SSR param, transforming the raw entry `body` to ignore `astro.config.mjs` configuration. The result is rendered using the `Renderer` component provided by `@astrojs/markdoc`:
```

Comment:
```suggestion
Astro recommends running the Markdoc transform step manually for this. This allows you to define your configuration and call Markdoc's rendering functions in a `.astro` file directly, bypassing `astro.config.mjs`.

You will need to install the `@markdoc/markdoc` package into your project.

```sh
# Using NPM
npx astro add @markdoc/markdoc
# Using Yarn
yarn astro add @markdoc/markdoc
# Using PNPM
pnpm astro add @markdoc/markdoc
```
Now, you can define Markdoc configuration options using `Markdock.transform()`.

This example defines an `abTestGroup` Markdoc variable based on an SSR param, transforming the raw entry `body`. The result is rendered using the `Renderer` component provided by `@astrojs/markdoc`:
```

I just felt the 2nd paragraph here was a little overloaded. So I tried to explain the what/why above, then leave this sentence to just explain exactly what is happening below. Something like this?

Reply:
Much better yes! 👏